src/lib/squid.ts

```diff
@@ -93,6 +94,21 @@ export const trackSkipTx = async ({
   return response.json();
 };
 
+const MAX_TRACK_TX_ATTEMPTS = 5;
+const TRACK_TX_INTERVAL = 1000;
+
+export const trackSkipTxWithTenacity = async ({
+  attemptNumber = 1,
+  ...skipParams
+}: SkipStatusParams & { attemptNumber?: number }) => {
+  if (attemptNumber === MAX_TRACK_TX_ATTEMPTS) return;
+  const { tx_hash: trackedTxHash } = await trackSkipTx(skipParams);
+  if (!trackedTxHash) {
+    sleep(TRACK_TX_INTERVAL);
```

Comment:
nit: would it make sense to just use setTimeout/setInterval directly?

Reply:
it would! i just felt like the implications of cleaning up timers felt like more work than making a recursive function tbh. but if it would be helpful for readability/updateability happy to make this change :)

Reply:
For now just going to merge as is but happy to update in folllowup!